libsigner/src/messages.rs

```diff
@@ -168,35 +219,126 @@ impl From<&RejectCode> for RejectCodeTypePrefix {
+                aggregate_key,
+                party_polynomials,
+            } => {
+                let party_polynomials: Vec<_> = party_polynomials
+                    .iter()
+                    .map(|(ix, commit)| (ix, commit.to_string()))
+                    .collect();
+                f.debug_struct("DkgResults")
+                    .field("aggregate_key", &aggregate_key.to_string())
+                    .field("party_polynomials", &party_polynomials)
+                    .finish()
+            }
+        }
+    }
 }
 
 impl SignerMessage {
     /// Helper function to determine the slot ID for the provided stacker-db writer id
-    pub fn msg_id(&self) -> u32 {
+    pub fn msg_id(&self) -> MessageSlotID {
         match self {
             Self::Packet(packet) => match packet.msg {
-                Message::DkgBegin(_) => DKG_BEGIN_MSG_ID,
-                Message::DkgPrivateBegin(_) => DKG_PRIVATE_BEGIN_MSG_ID,
-                Message::DkgEndBegin(_) => DKG_END_BEGIN_MSG_ID,
-                Message::DkgEnd(_) => DKG_END_MSG_ID,
-                Message::DkgPublicShares(_) => DKG_PUBLIC_SHARES_MSG_ID,
-                Message::DkgPrivateShares(_) => DKG_PRIVATE_SHARES_MSG_ID,
-                Message::NonceRequest(_) => NONCE_REQUEST_MSG_ID,
-                Message::NonceResponse(_) => NONCE_RESPONSE_MSG_ID,
-                Message::SignatureShareRequest(_) => SIGNATURE_SHARE_REQUEST_MSG_ID,
-                Message::SignatureShareResponse(_) => SIGNATURE_SHARE_RESPONSE_MSG_ID,
+                Message::DkgBegin(_) => MessageSlotID::DkgBegin,
+                Message::DkgPrivateBegin(_) => MessageSlotID::DkgPrivateBegin,
+                Message::DkgEndBegin(_) => MessageSlotID::DkgEndBegin,
+                Message::DkgEnd(_) => MessageSlotID::DkgEnd,
+                Message::DkgPublicShares(_) => MessageSlotID::DkgPublicShares,
+                Message::DkgPrivateShares(_) => MessageSlotID::DkgPrivateShares,
+                Message::NonceRequest(_) => MessageSlotID::NonceRequest,
+                Message::NonceResponse(_) => MessageSlotID::NonceResponse,
+                Message::SignatureShareRequest(_) => MessageSlotID::SignatureShareRequest,
+                Message::SignatureShareResponse(_) => MessageSlotID::SignatureShareResponse,
             },
-            Self::BlockResponse(_) => BLOCK_MSG_ID,
-            Self::Transactions(_) => TRANSACTIONS_MSG_ID,
+            Self::BlockResponse(_) => MessageSlotID::BlockResponse,
+            Self::Transactions(_) => MessageSlotID::Transactions,
+            Self::DkgResults { .. } => MessageSlotID::DkgResults,
+        }
+    }
+}
+
+impl SignerMessage {
+    /// Provide an interface for consensus serializing a DkgResults message
+    ///  without constructing the DkgResults struct (this eliminates a clone)
+    pub fn serialize_dkg_result<'a, W: Write, I>(
+        fd: &mut W,
+        aggregate_key: &Point,
+        party_polynomials: I,
+        write_prefix: bool,
```

Comment:
Minor nit: can you instead create separate methods for when `write_prefix` is either true or false?  It's confusing for me the reader to remember that the fourth argument to `serialize_dkg_result()` influences what the first byte of the message is going to be, since the `bool` value does not convey intent the way a method name would.

Reply:
Sure -- refactored so that the only public method serializes the full payload, and the internal method serializes the components.